Name: Really Silent Ringer Switch Mute
Version: 1.3.5.1-1
Author: Jason Robitaille
Description: Flicking the ringer switch to off will mute every aspect of the webOS, music and games included.  (MIT license)

--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js
@@ -2891,6 +2891,71 @@
 		method: 'status',
 		parameters: {"subscribe":true},
 		onSuccess: this.handleVoiceDialingAudioNotifications.bind(this)});
+		
+	this.controller.serviceRequest('palm://com.palm.audio/media', {
+		method:'status',
+		parameters: {},
+		onSuccess: function(response) {
+			this.setMutedSystem(!response['ringer switch']);
+			this.setMutedRingtone(!response['ringer switch']);
+			this.setMutedMedia(!response['ringer switch']);
+			this.setMutedPhone(!response['ringer switch']);
+		}.bind(this)
+	});
+	
+	this.controller.serviceRequest('palm://com.palm.keys/switches', {
+		method:'status',
+		parameters: {subscribe:true},
+		onSuccess: this.handleRingerNotifications.bind(this)
+	});
+},
+
+handleRingerNotifications: function(payload) {
+	if(payload.key=="ringer") {
+		if(payload.state=="up") { //unmute
+			this.setMutedSystem(false);
+			this.setMutedRingtone(false);
+			this.setMutedMedia(false);
+			this.setMutedPhone(false);
+		} else if(payload.state=="down") { //mute
+			this.setMutedSystem(true);
+			this.setMutedRingtone(true);
+			this.setMutedMedia(true);
+			this.setMutedPhone(true);
+		}
+	}
+},
+
+setMutedPhone: function(mute) {
+	this.controller.serviceRequest('palm://com.palm.audio/phone', {
+		method:'setMuted',
+		parameters:{muted:mute}
+	});
+},
+
+setMutedSystem: function(mute) {
+	this.controller.serviceRequest('palm://com.palm.audio/system', {
+		method:'setMuted',
+		parameters:{muted:mute}
+	});
+},
+
+setMutedRingtone: function(mute) {
+	this.controller.serviceRequest('palm://com.palm.audio/ringtone', {
+		method:'setMuted',
+		parameters:{muted:mute}
+	});
+},
+
+setMutedMedia: function(mute) {
+	var value = "up";
+	if(mute) {
+		value = "down";
+	}
+	this.controller.serviceRequest('palm://com.palm.audio/media', {
+		method:'ramp',
+		parameters:{command:value}
+	});
 },
 
 handlePhoneAudioNotifications: function(payload) {
